packages/headless/src/features/case-assist/case-assist-analytics-selectors.ts

```diff
@@ -134,13 +134,15 @@ export const caseAssistDocumentSuggestionSelector = (
 
   const result = {
     suggestionId: suggestion.uniqueId,
+    permanentId: suggestion.fields['permanentid'] as string,
```

Comment:
Should we have stronger typing for the `fields` property in the `DocumentSuggestionResponse` interface? This would prevent these random accesses and castings.

Right now it's typed as `Record<string, string | number | boolean | string[]>`, but I believe it could be:

```
fields: {
  permanentid: string;
  source: string;
  uri: string;
  urihash: string;
} & Record<string, string | number | boolean | string[]>;
```

instead.

Reply:
very good idea indeed!